Unit tests for helpers/config.js helpers/checkIpInList.js

reasonthearchitect opened the pull request:
** Note the following `-- Start of the original comment` is no longer true. It is kept as a record, however, after updating the package.json file to the latest version of `commander`, all command line switches work as expected.

** Note: Developers will need to run `npm install` after accepting this change as the old version of `commander` will fail.

`-- Start of the original comment`
Wrote some unit tests for the two files. Note that I did see some issues in the helpers/config file... as a result, some tests are missing. 

Non Functioning flags:
database: '-d' does not work, whereas, '--database' does.
httpPort: '--http-port' does not work, whereas '-h' does.
snapshot: '-s' does not work. '--snapshot' does.
config: '-c' no. '--config' yes.

Not sure if you want to have a bug on this... needs further exploration.

Isabello commented:
Just so we are clear, did you resolve these issue or are they still failing in the tests?

reasonthearchitect commented:
Isabello. To be clear.... I am trying to stay out of the prod code for now... until I get comfortable with the code base... hence all the unit tests...

So... as of now... I have left the src code as is as I was awaiting your suggestions. If you like I can dig into the code and see if I can fix it...

A few interesting things to consider... the flag -c does work from the command line... but not from the unit test.... I will take a peek if I can get the time today.

Isabello commented:
Ok so two options:

1. Resolve the issues in this PR
2. We accept the PR as is, but have the test set to skip so they can be enabled after we fix.

I would be ok with fixing the code as part of this PR, if you feel up to it. This section of code is not so sensitive compared to the rest of the code.

reasonthearchitect commented:
The following files changed:
- test/unit/helpers/config.js: Refactored the code so that all command line switches run successfully.
- package.json: No additional dependancies. Just update `commander` to the latest version.

Turns out the changes are very minor... Updated the version in the package.json file to the latest.

** Note: after this change developers need to run the `npm install` command; If not they will get an error.

Refactored the tests as the majority have the same setup, verify, and teardown functions.

Isabello commented on test/unit/helpers/config.js:
```diff
@@ -0,0 +1,193 @@
+	describe('config', function () {
+
+		it('single dash', function () {
+			process.argv.push('--config');
```
is this the correct test? it looks like the same as the `double dash` test

reasonthearchitect replied:
Nice catch. I have moved this test into the more generic test structure.

Isabello commented on test/unit/helpers/config.js:
```diff
@@ -0,0 +1,193 @@
+
+describe('config is able to override AppConfig properties from the command line', function () {
+
+	const VERSION = '0.0.0a';
```
We aren't on ES6 yet.

reasonthearchitect replied:
Noted.

reasonthearchitect commented:
Isabello. Thanks for taking the time to review the changes.
